valhalla/thor/pathalgorithm.h

```diff
@@ -32,8 +32,10 @@ class PathAlgorithm {
   /**
    * Constructor
    */
-  PathAlgorithm()
-      : interrupt(nullptr), has_ferry_(false), not_thru_pruning_(true), expansion_callback_() {
+  PathAlgorithm(uint32_t max_reserved_labels_count, bool clear_reserved_memory)
```

Comment:
imho, it's better to pass property tree into the constructor. in that case we would have common parameters parsing
```c++
config.get<uint32_t>("max_reserved_labels_count", kInitialEdgeLabelCountBD),
config.get<bool>("clear_reserved_memory", false)
```
only once here

Reply:
yeah good, remark, but how set `kInitialEdgeLabelCountBD` beautifully in `PathAlgorithm`? In different algos we set different `kInitialEdgeLabelCountBD`, for example for [Astar we set 1000000](https://github.com/valhalla/valhalla/blob/master/src/thor/bidirectional_astar.cc#L18) but for [multimodal we set 200000](https://github.com/valhalla/valhalla/blob/master/src/thor/multimodal.cc#L41)
@genadz

Reply:
we can update config by default value and forward it to `PathAlgoritm` as a copy, is it good?

Reply:
hm, I don't see a good way how to set different `max_reserved_labels_count` for different algorithms and keep parsing only in `PathAlgorithm`. so, probably for now it's okay to have constructor as you proposed
```c++
PathAlgorithm(uint32_t max_reserved_labels_count, bool clear_reserved_memory)
```
@kevinkreiser WDYT ?

Reply:
@genadz @keygang i agree, its completley fine to keep it that way until we think of something cleaner